.github/actions/build-prqlc/action.yaml

```diff
@@ -100,3 +100,7 @@ runs:
     - id: echo-artifact-name
       shell: bash
       run: echo "artifact-name=${{ env.ARTIFACT_NAME }}" >>"$GITHUB_OUTPUT"
+
+    - name: test-cli-works
+      shell: bash
+      run: ${{ env.ARTIFACT_NAME }} --help
```

Comment:
I agreed at first... But given the artifact is bundled, doesn't this require un-tarring / un-zipping? (not impossible but more than a line...)

Reply:
OK I'm trying for one target, which seems like it'll cover most failures

Reply:
Wouldn't it be better to test on all platforms to prevent something like #4268?
I was the one who added this in the first place, so I obviously should add tests for this......

Reply:
(By the way, it seems that prqlc CLI archives for macOS were broken from 0.11.2, same as #4254.)

Reply:
I added for mac and left a TODO for Windows!

Reply:
Wouldn't it be better to have a separate job for build, test and release and only release if the test is successful?

Reply:
I've not found "test first, then release if it passes" to be a good pattern — breaking a release isn't useful — the version is already tagged, we shouldn't be changing the code on a tag. Testing should be in previous runs, like we have here.

Open-minded on breaking up the job, I agree it's a bit of a bolt-on atm...